dev/io.openliberty.microprofile.openapi.ui.internal_fat/fat/src/io/openliberty/microprofile/openapi/ui/internal/fat/tests/UICustomPathTest.java

```diff
@@ -0,0 +1,159 @@
+                    Props.getInstance().getFileProperty(Props.DIR_LOG),
+                    VncRecordingFormat.MP4)
+            .withLogConsumer(new SimpleLogConsumer(UICustomPathTest.class, "selenium-driver"));
+
+    private RemoteWebDriver driver;
+
+    @BeforeClass
+    public static void setup() throws Exception {
+        WebArchive war = ShrinkWrap.create(WebArchive.class, APP_NAME + ".war")
+                .addClass(TestResource.class);
+
+        ShrinkHelper.exportDropinAppToServer(server, war, DeployOptions.SERVER_ONLY);
+
+        ServerFileUtils serverFileUtils = new ServerFileUtils();
+
+        String bootProps = serverFileUtils.getServerFileLoc(server) + "/bootstrap.properties";
+        // append to bootstrap.properties
+        FileWriter writer = new FileWriter(bootProps, true);
+        writer.append(PROPERTY_NAME + "=" + PATH_VALUE);
+        writer.append(System.getProperty("line.separator"));
+        writer.close();
+
+        String jvmOptions = serverFileUtils.getServerFileLoc(server)+"/jvm.options";
+        writer = new FileWriter(jvmOptions, true);
+        writer.append("-Dcom.ibm.ws.beta.edition" + "=" + "true");
+        writer.close();
+
+        server.startServer();
+
+        Testcontainers.exposeHostPorts(server.getHttpDefaultPort(), server.getHttpDefaultSecurePort());
+    }
+
+    @Before
+    public void setupTest() {
+        driver = new RemoteWebDriver(chrome.getSeleniumAddress(), new ChromeOptions());
+    }
+
+    @Test
+    public void testCustomUIPath() {
+        driver.get("http://host.testcontainers.internal:" + server.getHttpDefaultPort() + PATH_VALUE);
+
+        // Check the title loads
+        WebElement title = waitForElement(driver, By.cssSelector("h2.title"), LONG_WAIT);
+        assertThat("Page title", title.getText(), Matchers.containsString("Generated API"));
+
+        // Check the headerbar colour
+        WebElement headerbar = waitForElement(driver, By.cssSelector("div.headerbar"));
+        assertEquals("Headerbar colour", "rgba(25, 28, 44, 1)", headerbar.getCssValue("background-color"));
+
+        // Check the headerbar has a background image. It's a data URL, so it's hard to assert it's actually correct, we just assert that there is one
+        WebElement headerbarWrapper = waitForElement(headerbar, By.cssSelector("div.headerbar-wrapper"));
+        assertThat("Headerbar image", headerbarWrapper.getCssValue("background-image"), startsWith("url(\"data:image/png"));
+
+        // Check we can see and open the operation
+        WebElement testGetOpBlock = waitForElement(driver, By.id("operations-default-get_test"));
+        WebElement testGetButton = testGetOpBlock.findElement(By.tagName("button"));
+        testGetButton.click();
+        WebElement testGet200Response = waitForElement(testGetOpBlock, By.cssSelector("tr.response[data-code=\"200\"]"));
+        assertNotNull("200 response line", testGet200Response);
+    }
```

Comment:
We could probably make this code common as well if we're going to want to repeat it in multiple places.

After calling `driver.get`, the only thing that would need to be passed in is the `driver`.